frank-flow/src/frontend/src/app/flow/flow.component.ts

```diff
@@ -69,18 +97,73 @@ export class FlowComponent implements AfterViewInit {
     this.changeCanvasSize('height', this.canvasExpansionSize);
   }
 
-  changeCanvasSize(
-    direction: 'height' | 'width',
+  setBasicCanvasSize(): void {
+    this.calculateMinimumCanvasSize();
+
+    this.renderCanvas('width', this.minimumXPosition + this.nodeBufferSpace);
+    this.renderCanvas('height', this.minimumYPosition + this.nodeBufferSpace);
+  }
+
+  changeCanvasSize(direction: canvasDirection, expansionValue: number): void {
+    if (this.canChangeCanvasSize(direction, expansionValue)) {
+      this.renderCanvas(
+        direction,
+        this.getNewCanvasSize(direction, expansionValue)
+      );
+    }
+  }
+
+  renderCanvas(direction: canvasDirection, value: number): void {
+    this.renderer.setStyle(this.canvasElement, direction, value + 'px');
+  }
+
+  canChangeCanvasSize(
+    direction: canvasDirection,
     expansionValue: number
-  ): void {
-    this.renderer.setStyle(
-      this.canvasElement,
-      direction,
-      (direction == 'height'
+  ): boolean {
+    this.calculateMinimumCanvasSize();
+    if (direction === 'height') {
+      return (
+        this.getNewCanvasSize(direction, expansionValue) > this.minimumYPosition
+      );
+    } else {
+      return (
+        this.getNewCanvasSize(direction, expansionValue) > this.minimumXPosition
+      );
+    }
+  }
+
+  getNewCanvasSize(direction: canvasDirection, expansionValue: number): number {
+    return (
+      (direction === 'height'
         ? this.canvasElement.offsetHeight
-        : this.canvasElement.offsetWidth) +
-        expansionValue +
-        'px'
+        : this.canvasElement.offsetWidth) + expansionValue
     );
   }
```

Comment:
If canvasElement is going to be HTMLElement it can be undefined. So there needs to be a check:
```TypeScript
getNewCanvasSize(direction: canvasDirection, expansionValue: number): number {
    return ((
        (direction === 'height'
          ? this.canvasElement?.offsetHeight
          : this.canvasElement?.offsetWidth) ?? 0) + expansionValue
    );
  }
```